rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1578,19 +1578,32 @@ public void addSignature(BtcECKey federatorBtcPublicKey, List<byte[]> signatures
 
         Context.propagate(btcContext);
 
+        if (svpIsOngoing()) {
```

Comment:
Maybe we could wrap everything in a single svp method?

```
if (svpIsOngoing()) {
    addSignatureForSvp();
   return;
} 
...
// Normal addSignature logic
```
To keep simple and focused, while at the same time re-using addSignature method. What do you think?

Reply:
I agree with @marcos-iov  's proposal. I think separating these two logic into two methods makes much sense.

Reply:
that way you won't sign a normal pegout when svp is ongoing :)

Reply:
i thought about doing something like that with a `continue` instead of a `return`, but i discarded it since (correct me if im wrong) it will make the flow continue after signing the spend tx.
Anyways thats no big deal, because it will return early enough [here](https://github.com/rsksmart/rskj/blob/6c2db8ccc7125ef861da7f4df912ca304bcbdffd/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L1597)